test/unittests/stack_test.cpp

```diff
@@ -48,3 +52,95 @@ TEST(stack, drop_and_peek)
     EXPECT_EQ(stack.size(), 0);
     EXPECT_TRUE(stack.empty());
 }
+
+TEST(stack, clear)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    stack.drop(0);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    stack.clear();
+    EXPECT_TRUE(stack.empty());
+    EXPECT_EQ(stack.size(), 0);
+}
+
+TEST(stack, resize)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 2);
+
+    // grow stack
+    stack.resize(4);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.peek(), 0);
+    EXPECT_EQ(stack.peek(1), 0);
+    EXPECT_EQ(stack.size(), 4);
+
+    stack.drop();
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 3);
+
+    // shrink stack
+    stack.resize(1);
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 1);
+
+    stack.clear();
+    EXPECT_TRUE(stack.empty());
+}
+
+TEST(stack, iterator)
+{
+    Stack<char> stack;
+    stack.push('a');
+    stack.push('b');
+    stack.push('c');
+
+    EXPECT_FALSE(stack.empty());
+    EXPECT_EQ(stack.size(), 3);
+
+    auto bottom_item = stack.begin();
```

Comment:
BTW `cend()` should be enough in the code currently using `end()`, maybe makes sense to change it.

Otherwise this test should check also that iterators allow to change elements.

Reply:
> Otherwise this test should check also that iterators allow to change elements.

You mean something like what's in 61a86f41?

Reply:
Yeah that would do.
But not sure if we should allow this if we create another implementation of stack.

Reply:
I think it is fine now, at least we have all the features used tested. If there's a follow up PR to remove the use of `end` and replace it with `cend` or to remove those use cases entirely, then we can remove this.